superflore/docker.py

```diff
@@ -69,25 +71,45 @@ def login(self):
     def pull(self, org, repo, tag='latest'):
         self.image = self.client.images.pull('%s/%s:%s' % (org, repo, tag))
 
-    def run(self, rm=True, show_cmd=False, privileged=False):
-        cmd_string = "bash -c '"
-        for i, bash_cmd in enumerate(self.bash_cmds):
-            cmd_string += bash_cmd
-            if i != len(self.bash_cmds) - 1:
-                cmd_string += ' && '
-        cmd_string += "'"
-        if show_cmd:
-            msg = "Running container with command string '%s'..."
-            info(msg % cmd_string)
+    def get_log(self):
+        return self.log
+
+    def get_command(self, logging_dir=None):
+        if logging_dir:
+            cmd = "bash -c '"
+            cmd += (" &>> %s/log.txt && " % logging_dir).join(self.bash_cmds)
+            cmd += (" &>> %s/log.txt'" % logging_dir)
+        else:
+            cmd = "bash -c '" + " && ".join(self.bash_cmds) + "'"
+        return cmd
 
-        self.client.containers.run(
-            image=self.image,
-            remove=rm,
-            command=cmd_string,
-            privileged=privileged,
-            volumes=self.directory_map,
-        )
-        ok("Docker container exited.")
+    def run(self, rm=True, show_cmd=False, privileged=False):
+        with TempfileManager(None) as tmp:
+            # change access to the directory
+            os.chmod(tmp, 17407)
+            # map into the container
+            self.map_directory(tmp)
+            cmd_string = self.get_command(tmp)
+            if show_cmd:
+                msg = "Running container with command string '%s'..."
+                info(msg % self.get_command())
```

Comment:
The issue with that is the fact that `get_command()` without a log file will return the list, but, otherwise, shows a `&&> /tmp/[tmp directory/log.txt` file.

I could perhaps just do a `.replace`, but I don't think it's any more efficient at that point.

Or, if you want, I can just have it show the temporary log file mapping.

Reply:
I think it's better to show the real values. If you mention the mounting right after that would allow people to debug easily.

Reply:
Will do